Havit.Blazor.Components.Web.Bootstrap/TreeViews/Internal/HxTreeViewItemInternal.razor.cs

```diff
@@ -18,16 +18,82 @@ public partial class HxTreeViewItemInternal<TItem> : ComponentBase
 
 		[CascadingParameter] protected HxTreeView<TItem> TreeViewContainer { get; set; }
 
+		protected HxCollapse collapseComponent;
+		/// <summary>
+		/// Indicates whether the collapse is currently animating (expanding or collapsing).
+		/// </summary>
+		protected bool animating;
+
+		protected bool initiallyExpanded;
+
+		protected override void OnInitialized()
+		{
+			initiallyExpanded = this.InitialExpandedSelector?.Invoke(this.Item) ?? false;
+			if (initiallyExpanded)
+			{
+				IsExpanded = true;
+			}
+		}
+
 		private async Task HandleItemClicked()
 		{
 			await this.OnItemSelected.InvokeAsync(this.Item);
 		}
 
-		private Task HandleItemExpanderClicked()
+		private async Task HandleItemExpanderClicked()
+		{
+			if (this.IsExpanded.GetValueOrDefault())
+			{
+				await Collapse();
+			}
+			else
+			{
+				await Expand();
+			}
+		}
+
+		/// <summary>
+		/// Expand the item with an animation.
+		/// </summary>
+		/// <returns></returns>
+		public async Task Expand()
```

Comment:
The names should be `ExpandAsync` and `CollapseAsync`.
Do we want to expose these methods as `public`? Do we call them from outside?
(Let's keep the internal components as encapsulated as possible.)

Reply:
I was thinking it may come in handy if the user wanted to for example create the functionality `InitiallyExpanded` provides but with the animation. I've made them `protected` for now, let me know if you find that ok.

Reply:
If we will want to use it, we will change it to anything more accessible later. Internal components are not meant to be used by any user in any scenario. They are only meant to be used in Havit.Blazor library itself. They are `public` only for the reason the `Component` has to be public to be able to use it in `.razor` markup.

Reply:
Ok, will change it to `private`.